Automattic/wp-calypso
# Social previews | Fix SIG and Custom media not used

Right now the Social Previews in Calypso are unaware of SIG images or custom media selected in share post section. This PR aims to fix that.

## Proposed Changes

* Enable SIG and custom media support for previews
* Update preview wrappers to pass custom media
* Fix empty Twitter preview when no text/description is provided

## Testing Instructions



* Boot up this Jetpack PR https://github.com/Automattic/jetpack/pull/33504
* Create a post via wp-admin
* Add a custom media
* Publish the post
* Now boot up this PR
* Goto `http://calypso.localhost:3000/posts/:site`
* For the newly published post for the above site, click on the 3 dots and select Share
* Click on Preview
* Confirm that you see the custom media
* Try the above for both image and a video
* Now enable SIG for the post and save
* Reload the above Calypso page 
* Open the preview again
* Confirm that SIG image is shown in the preview

<table>
    <thead>
        <tr>
            <th>BEFORE</th>
            <th>AFTER</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <th colspan="2">Custom video</th>
        </tr>
        <tr>
            <td>
                <img width="1333" alt="Screenshot 2023-10-09 at 7 07 07 PM"
                    src="https://github.com/Automattic/wp-calypso/assets/18226415/ead003ff-0aa7-481a-b7e1-0c260a631295">
            </td>
            <td>
                <img width="1338" alt="Screenshot 2023-10-09 at 6 47 01 PM"
                    src="https://github.com/Automattic/wp-calypso/assets/18226415/a26513b4-a982-4e97-b160-4b2e083536a1">
            </td>
        </tr>
        <tr>
            <th colspan="2">Custom image</th>
        </tr>
        <tr>
            <td>
                <img width="1340" alt="Screenshot 2023-10-09 at 7 05 13 PM"
                    src="https://github.com/Automattic/wp-calypso/assets/18226415/86524049-4607-4ae7-a641-6c25edf9bf2b">
            </td>
            <td>
                <img width="1336" alt="Screenshot 2023-10-09 at 7 04 10 PM"
                    src="https://github.com/Automattic/wp-calypso/assets/18226415/fcffb2e4-e0d7-4839-8891-add7f5a5bfc4">
            </td>
        </tr>
        <tr>
            <th colspan="2">SIG image</th>
        </tr>
        <tr>
            <td>
                <img width="1339" alt="Screenshot 2023-10-09 at 7 07 33 PM"
                    src="https://github.com/Automattic/wp-calypso/assets/18226415/c86165bf-117d-496f-bf4d-d836bb94557f">
            </td>
            <td>
                <img width="1339" alt="Screenshot 2023-10-09 at 6 47 48 PM"
                    src="https://github.com/Automattic/wp-calypso/assets/18226415/336d0f6f-0012-4ae1-9f43-962219389cf5">
            </td>
        </tr>
</table>


## Pre-merge Checklist
